clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1147,8 +1147,9 @@ public boolean shouldBlock() {
      *             this function is called
      * @throws org.apache.kafka.common.errors.AuthorizationException if not authorized to the topic or to the
      *             configured groupId
+     * @throws java.lang.IllegalArgumentException if the committed offset is negative
```

Comment:
Not sure it's necessary to document this since the offsets to commit are derived internally and should be correct.